src/components/views/dialogs/ChatInviteDialog.js

```diff
@@ -435,6 +445,39 @@ module.exports = React.createClass({
         return inviteList;
     },
 
+    _lookupThreepid(medium, address) {
+        // wait a bit to let the user finish typing
+        return q.delay(500).then(() => {
+            // If the query has changed, forget it
+            if (this.state.queryList[0] && this.state.queryList[0].address !== address) {
```

Comment:
Done, but note you can't remove `cancelThreepidLookup` from `this` in the finally because you don't know that it's the same one you added.